### PR TITLE
fix firefox sync-server

### LINKED ISSUE #38830: Syncserver compilation failed
## Issue description
There is a problem when Nixos compile the serversyncstorage from mozilla. The problem is that syncserver use method .app of WSGIProxy2 but this method isn't available before 0.3

I will start to modified requirements.txt of mozservices and when it will be release, it would be great to change the version in : 
https://github.com/NixOS/nixpkgs/blob/7ebb82e04f17d96313f1d60c748f32a7f7bee9f6/pkgs/top-level/python-packages.nix#L10496

### Steps to reproduce
Enable syncserver in Nixos

You will have this output : 
https://hydra.nixos.org/build/72124841/log 

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.32, NixOS, 18.03.131807.489a14add9a (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0`
 - channels(root): `"nixos-18.03.131807.489a14add9a"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

```

### PR DESCRIPTION
**Pull request by tg-x**
###### Motivation for this change

`pythonPackages.serversyncstorage` has been broken,
this PR fixes it by disabling tests and removing a conflicting WSGIProxy dependency only used by tests

tested running `services.firefox.syncserver`, the service starts and shows the 'it works' status page

fixes #38830

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/syncserver/default.nix [hunk 1]
```diff
@@ -16,13 +16,13 @@
...
   disabled = ! isPy27;
 
   src = fetchgit {
     url = https://github.com/mozilla-services/syncserver.git;
     rev = "refs/tags/${version}";
-    sha256 = "1fsiwihgq3z5b5kmssxxil5g2abfvsf6wfikzyvi4sy8hnym77mb";
+    sha256 = "0hxjns9hz7a8r87iqr1yfvny4vwj1rlhwcf8bh7j6lsf92mkmgy8";
   };
 
   buildInputs = [ unittest2 ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/syncserver/default.nix]: `checkInputs`?

**Review by dotlambda — CHANGES_REQUESTED** [review]
Two packages are missing a `meta` section. Please add one.
Also, add a `pythonPackages.` prefix to the commit messages.

File: pkgs/development/python-modules/serversyncstorage/default.nix [hunk 2]
```diff
@@ -13,29 +11,40 @@
...
     pyramid sqlalchemy simplejson mozsvc cornice pyramid_hawkauth pymysql
-    pymysqlsa umemcache WSGIProxy requests pybrowserid
+    pymysqlsa umemcache requests pybrowserid
+    webtest
   ];
 
-  meta = {
-    broken = true; # 2018-11-04
-  };
+  patchPhase = ''
```
Comment (FRidh) [inline pkgs/development/python-modules/serversyncstorage/default.nix]: postPatch

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/serversyncstorage/default.nix [hunk 3]
```diff
@@ -13,29 +11,40 @@
...
+    sed -i "s/WSGIProxy.*//" requirements.txt
+    sed -i "s/'wsgiproxy',//" setup.py
+
+    # since we disabled tests (which would need WSGIProxy) lets get rid of the rest of the test dependencies as well:
+    sed -i "s/unittest2.*//" requirements.txt
+    sed -i "s/'unittest2',//" setup.py
+    sed -i "s/webtest.*//" requirements.txt
+    sed -i "s/'webtest',//" setup.py
+    sed -i "s/testfixtures.*//" requirements.txt
+    sed -i "s/, 'testfixtures'//" setup.py
```
Comment (dotlambda) [inline pkgs/development/python-modules/serversyncstorage/default.nix]: I'd prefer using `substituteInPlace`. That way, you'll also have a single command for every file.

**Comment by tg-x** [comment]
applied the suggested changes

File: pkgs/development/python-modules/serversyncstorage/default.nix (hunk 3 above)
Comment (tg-x) [reply pkgs/development/python-modules/serversyncstorage/default.nix]: sed also does regexp which is used in this case, I merged it now into one command per file

File: pkgs/development/python-modules/serversyncstorage/default.nix [hunk 4]
```diff
@@ -13,29 +12,46 @@
...
+    rev = version;
+    sha256 = "08xclxj38rav8yay9cijiavv35jbyf6a9jzr24vgcna8pjjnbbmh";
   };
 
-  checkInputs = [ testfixtures unittest2 webtest ];
   propagatedBuildInputs = [
     pyramid sqlalchemy simplejson mozsvc cornice pyramid_hawkauth pymysql
-    pymysqlsa umemcache WSGIProxy requests pybrowserid
+    pymysqlsa umemcache requests pybrowserid
+    webtest
```
Comment (dotlambda) [inline pkgs/development/python-modules/serversyncstorage/default.nix]: If `webtest` is a test dependency, why is it here anyway?

File: pkgs/development/python-modules/serversyncstorage/default.nix [hunk 5]
```diff
@@ -13,29 +12,46 @@
...
   ];
 
-  meta = {
-    broken = true; # 2018-11-04
+  postPatch = ''
+    # conflict between WSGIProxy & WSGIProxy2 required by webtest
+    # (which is required by pyramid_hawkauth)
```
Comment (dotlambda) [inline pkgs/development/python-modules/serversyncstorage/default.nix]: This comment does not make sense

File: pkgs/development/python-modules/mozsvc/default.nix [hunk 6]
```diff
@@ -1,32 +1,27 @@
...
-    url = https://github.com/nbp/mozservices/commit/f86c0b0b870cd8f80ce90accde9e16ecb2e88863.diff;
-    sha256 = "1lnghx821f6dqp3pa382ka07cncdz7hq0mkrh44d0q3grvrlrp9n";
-  });
-
   doCheck = false; # lazy packager
   propagatedBuildInputs = [ pyramid simplejson konfig ];
 
   meta = with stdenv.lib; {
-    homepage = https://github.com/mozilla-services/mozservices;
+    homepage = "https://github.com/mozilla-services/mozservices";
```
Comment (dotlambda) [inline pkgs/development/python-modules/mozsvc/default.nix]: no quotes

File: pkgs/development/python-modules/syncserver/default.nix [hunk 7]
```diff
@@ -16,18 +17,26 @@
...
+  checkInputs = [ unittest2 ];
   propagatedBuildInputs = [
     cornice gunicorn pyramid requests simplejson sqlalchemy mozsvc tokenserver
     serversyncstorage configparser
   ];
+
+  meta = with stdenv.lib; {
+    description = "Run-Your-Own Firefox Sync Server";
+    homepage = "https://github.com/mozilla-services/syncserver";
+    platforms = platforms.unix;
```
Comment (dotlambda) [inline pkgs/development/python-modules/syncserver/default.nix]:
```suggestion
```

File: pkgs/development/python-modules/serversyncstorage/default.nix [hunk 8]
```diff
@@ -13,29 +12,46 @@
...
+    sed -i "s/unittest2.*//; s/webtest.*//; s/testfixtures.*//" requirements.txt
+    sed -i "s/'unittest2',//; s/'webtest',//; s/, 'testfixtures'//" setup.py
+  '';
+
+  doCheck = false;
+
+  meta = with stdenv.lib; {
+    description = "The SyncServer server software, as used by Firefox Sync";
+    homepage = "https://github.com/mozilla-services/server-syncstorage";
+    platforms = platforms.unix;
```
Comment (dotlambda) [inline pkgs/development/python-modules/serversyncstorage/default.nix]:
```suggestion
```
This is set automatically by `buildPython*`.

File: pkgs/development/python-modules/serversyncstorage/default.nix [hunk 9]
```diff
@@ -13,29 +12,46 @@
...
-    broken = true; # 2018-11-04
+  postPatch = ''
+    # conflict between WSGIProxy & WSGIProxy2 required by webtest
+    # (which is required by pyramid_hawkauth)
+    sed -i "s/WSGIProxy.*//" requirements.txt
+    sed -i "s/'wsgiproxy',//" setup.py
+
+    # since we disabled tests (which would need WSGIProxy)
+    # lets get rid of the rest of the test dependencies as well:
+    sed -i "s/unittest2.*//; s/webtest.*//; s/testfixtures.*//" requirements.txt
```
Comment (dotlambda) [inline pkgs/development/python-modules/serversyncstorage/default.nix]: I don't see a need for patching `requirements.txt`. Then you can use `substituteInPlace` after all.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by Nadrieril** [comment]
I have tried to run it on my server, and it compiles, but running `syncserver` crashes with `pyramid.exceptions.ConfigurationError: 'acl' is not supported`. This exception is raised from the `cornice` package, which has version `3.4.2` in nixpkgs but is required version `0.17` from `serversyncstorage`.

**Comment by Nadrieril** [comment]
I have made a PR that fixes this: https://github.com/NixOS/nixpkgs/pull/54739.
I have tested it in production and it seems to work well.

**Pull request closed without merging**